ui/src/main/java/org/pentaho/di/ui/core/database/wizard/CreateDatabaseWizardPage1.java

```diff
@@ -228,17 +231,33 @@ public void setAccessTypes() {
 
   public boolean canFlipToNextPage() {
     String name = wName.getText() != null ? wName.getText().length() > 0 ? wName.getText() : null : null;
-    String dbType = wDBType.getSelection().length == 1 ? wDBType.getSelection()[0] : null;
-    String acType = wAccType.getSelection().length == 1 ? wAccType.getSelection()[0] : null;
+    String dbType = wDBType.getSelection().length == 1 ? wDBType.getSelection()[ 0 ] : null;
+    String acType = wAccType.getSelection().length == 1 ? wAccType.getSelection()[ 0 ] : null;
 
     if ( name == null || dbType == null || acType == null ) {
       setErrorMessage( BaseMessages.getString( PKG, "CreateDatabaseWizardPage1.ErrorMessage.InvalidInput" ) );
       return false;
     }
     if ( name != null && DatabaseMeta.findDatabase( databases, name ) != null ) {
-      setErrorMessage( BaseMessages.getString( PKG, "CreateDatabaseWizardPage1.ErrorMessage.DBNameExists",
-        name.trim() ) );
-      return false;
+      try {
+        DatabaseManagementInterface dbMgr =
+          Spoon.getInstance().getBowl().getManager( DatabaseManagementInterface.class );
+        if ( dbMgr.getDatabases().stream().anyMatch( db -> db.getName().equals( name ) ) ) {
```

Comment:
getDatabase( name ) would be far more efficient here.

Reply:
fixed